src/ui/components/SpecialDistribution.js

```diff
@@ -41,88 +76,118 @@ export const SpecialDistribution = () => {
-            onChange={(e) => setMemo(e.currentTarget.value)}
-          />
-        </p>
-        <p>
-          <label htmlFor="timestamp">Timestamp</label> <br />
-          <input
-            type="number"
-            name="timestamp"
-            value={credTimestamp}
-            onChange={(e) => {
-              const newTimestamp = +e.currentTarget.value;
-              if (!Number.isFinite(newTimestamp)) {
-                throw new Error(`invalid timestamp: ${e.currentTarget.value}`);
-              }
-              setCredTimestamp(newTimestamp);
-            }}
+      {successMessage && (
+        <div className={classes.centerRow}>
+          <Alert severity="success" className={classes.element}>
+            {successMessage}
+          </Alert>
+        </div>
+      )}
+      <div className={classes.centerRow}>
+        <div className={`${classes.centerRow} ${classes.element}`}>
+          <div className={classes.arrowBody}>
+            <TextField
+              className={classes.arrowInput}
+              size="medium"
+              InputProps={{disableUnderline: true}}
+              type="number"
+              placeholder="(amount)"
+              value={amount}
+              onChange={(e) => setAmount(e.currentTarget.value)}
+            />
+          </div>
+          <div className={classes.triangle} />
+        </div>
+        <div
+          className={`${classes.dropdownWrapper} ${classes.centerRow} ${classes.element}`}
+        >
+          <AccountDropdown
+            ledger={ledger}
+            setCurrentAccount={setRecipient}
+            placeholder="Recipient"
           />
-        </p>
-        <input
-          disabled={recipient == null}
-          type="submit"
-          value="distribute special grain"
+        </div>
+      </div>
+      <div className={classes.centerRow}>
+        <TextField
+          variant="outlined"
+          size="medium"
+          className={classes.element}
+          type="text"
+          value={memo}
+          placeholder="Memo"
+          onChange={(e) => setMemo(e.currentTarget.value)}
         />
-        <br />
-        <input
-          type="button"
-          value="save ledger to disk"
-          onClick={() => {
-            fetch("data/ledger.json", {
-              headers: {
-                Accept: "text/plain",
-                "Content-Type": "text/plain",
-              },
-              method: "POST",
-              body: ledger.serialize(),
-            });
+        <TextField
+          variant="outlined"
+          size="medium"
+          className={classes.element}
+          type="number"
+          value={credTimestamp}
+          placeholder="Timestamp"
+          onChange={(e) => {
+            const newTimestamp = +e.currentTarget.value;
+            if (!Number.isFinite(newTimestamp)) {
+              throw new Error(`invalid timestamp: ${e.currentTarget.value}`);
+            }
+            setCredTimestamp(newTimestamp);
```

Comment:
A couple thoughts about this component:

1. Do we need it? I suggest getting rid of it above, assuming this component is to be used to create a user-generated timestamp. Is a user-modifiable timestamp a desired feature here?
2. Assuming the answer to 1 is yes, let's allow users to configure this using a date picker of some sort. I personally struggle to eyeball unix epoch timestamps, and I imagine most other people to do.

Reply:
I'm all for removing it, and just setting the timestamp in the submit function, but it's up to @decentralion

Reply:
Users might want to do special distributions in the past. e.g. For MetaGame, we would want to distribute SEEDs in the ledger based on when we did our distribution on chain a few months ago.

Reply:
According to conversation in discord, it seems this should stay